### ISSUE TITLE
train_mod_flipflop.py warns about over half of chunks failing maxdwell

### ISSUE THREAD

**Issue by lpryszcz** [nanoporetech/taiyaki#22]
Hi, I've started training model for RNA modification detection, but train_mod_flipflop.py warns about over half of chunks failing maxdwell. 
```
* Summary: pass:100 maxdwell:112 tooshort:8 emptysequence:3 meandwell:23
...* Warning: only 126 chunks passed tests after 278 attempts.
* Summary: pass:126 maxdwell:111 tooshort:7 meandwell:29 emptysequence:5
.* Warning: only 111 chunks passed tests after 272 attempts.
* Summary: pass:111 emptysequence:10 maxdwell:114 meandwell:30 tooshort:7
41 0.103  179.49s (128.74 ksample/s 4.55 kbase/s) lr=9.96e-04 pass:332681 maxdwell:194491  tooshort:13425  meandwell:60513  emptysequence:20365
```

When I played with test data provided with taiyaki less than 10% of chunks failed maxdwell. I suspect some modification may in fact affect helicase thus dwell time may increase significantly and I suspect that's what's happening... 
```
.................................................C  1000 0.029  151.65s (66.01 ksample/s 6.76 kbase/s) lr=4.00e-04 pass:3468812  meandwell:207586  maxdwell:342531  emptysequence:7256
```
Would it be reasonable to increase the values of `--filter_max_dwell` and `--filter_mean_dwell`? If so, how to estimate which values would be good (defaults are 10 and 3, respectively)?

**Comment by aevansNP** [comment]
Hello - yes, this would be a reasonable thing to try.
It could be that your data just has a wide variation of dwell times.

It might also be worth investigating further. There is a script in the misc directory which will plot the mappings in your training data so that you can see (for example) whether reads are being rejected because of a single very long dwell, or because there are lots of long dwells in every read.

You can do (for example)

taiyaki/misc/plot_mapped_signals.py --output map_plot.png --nreads 100 your_mapped_signal_file.hdf5

to plot the first 100 mapped reads in your mapped signal file. There are also command-line options for the script which will allow you to zoom in on a particular region.

**Comment by lpryszcz** [comment]
Thanks a lot for prompt answer, @aevansNP . 
I have two quick questions: 
- How many reads do you recommend for training? I see that with 16k reads the training will take ages... Would 4k do the job? 
- What would be reasonable ratio of modified to unmodified reads?

**Comment by aevansNP** [comment]
---How many reads do you recommend for training?
The answer to that question depends a lot on the details - but at least  for DNA, we've found that you can train a reasonable model with a few thousand reads. More data does tend to lead to models that generalise better, though.
The training time shouldn't depend on the number of reads (except for a bit of overhead for reading very large hdf5 files), although the time for remapping will, of course.

---What would be a reasonable ratio of modified to unmodified reads?
Hard to answer this without knowing the details of what you are doing (which we probably don't want to get into). Of course trained models (if everything goes well) will tend to reflect the data that they are trained on, so if the fraction of mods is too small then the model is likely to ignore them.

When training models to call mod bases, we typically use training data where all the reads have some modifications, but in any read the density of modifications is quite low.

**Comment by lpryszcz** [comment]
Thanks, you're being really helpful @aevansNP !  Then I won't downsample my read repository and try to focus on reads having at least one modification present. 

I've set `--min_batch_size 100 --filter_max_dwell 15` and training seems running smooth now. Interestingly, the process is much faster (at least 10x) when most chunks are being included. Any idea why is that? 
```
..................................................    60 0.094  97.86s (102.21 ksample/s 3.27 kbase/s) lr=9.92e-04 pass:206397  maxdwell:44893  meandwell:16210  emptysequence:9285  tooshort:6151 
```

Also, I got peculiar observation regarding runtime. Training process on my [GTX 1660Ti](https://www.techpowerup.com/gpu-specs/geforce-gtx-1660-ti.c3364) (6GB vRAM --min_batch_size 50: 104.37 ksample/s 3.35 kbase/s) reports theoretically slower learning than in [GTX 1080 Ti](https://www.techpowerup.com/gpu-specs/geforce-gtx-1080-ti.c2877) (11 GB vRAM, --min_batch_size 100: 122.85 ksample/s 3.89 kbase/s) that's installed in the sever on the same data. However 1660Ti went through 60 iterations, while 1080Ti less than 35 in the same time interval. Both machines are not under load and have same libraries (CUDA 10.1, driver v418..67) - it's weird, right? 
1660Ti has slightly higher GPU clock (only by the margin) and slightly faster memory (GDDR6 vs GDDR5X), still 1080Ti memory has much higher memory bandwidth, so such big difference in speed doesn't really make sense to me, unless there is some CPU bound component (my local CPU is way faster than the server one)! Am I missing something?
